store/tikv/coprocessor.go

```diff
@@ -545,27 +630,33 @@ func (it *copIterator) open(ctx context.Context) {
 				minCommitTSPushed: &it.minCommitTSPushed,
 				Client:            it.store.client,
 			},
-
-			memTracker: it.memTracker,
-
+			memTracker:      it.memTracker,
 			replicaReadSeed: it.replicaReadSeed,
 		}
 		go worker.run(ctx)
 	}
+	bo := NewBackoffer(ctx, copBuildTaskMaxBackoff).WithVars(it.vars)
+	_, err := buildCopTasksChan(bo, it.store.regionCache,
+		&copRanges{mid: it.req.KeyRanges}, it.req, newTaskCh, it.finishCh)
+	if err != nil {
+		return err
+	}
 	taskSender := &copIteratorTaskSender{
-		taskCh:   taskCh,
-		wg:       &it.wg,
-		tasks:    it.tasks,
-		finishCh: it.finishCh,
-		sendRate: it.sendRate,
+		tasksFromBuilderCh: newTaskCh,
+		taskCh:             taskCh,
+		wg:                 &it.wg,
+		finishCh:           it.finishCh,
+		sendRate:           it.sendRate,
+		respChan:           it.respChan,
+		orderedTasksCh:     it.tasks,
 	}
-	taskSender.respChan = it.respChan
 	go taskSender.run()
+	return nil
 }
 
 func (sender *copIteratorTaskSender) run() {
 	// Send tasks to feed the worker goroutines.
-	for _, t := range sender.tasks {
+	for t := range sender.tasksFromBuilderCh {
```

Comment:
This chanel should not be blocked when,
1. `buildCopTasksChan` meets an error
2. `buildCopTasksChan` get an exit signal from `finishCh`

And `copIterator.Next` should return error if `buildCopTasksChan` meets an error.